homeassistant/components/iometer/coordinator.py

```diff
@@ -58,4 +67,18 @@ async def _async_update_data(self) -> IOmeterData:
         except IOmeterConnectionError as error:
             raise UpdateFailed(f"Error communicating with IOmeter: {error}") from error
 
+        fw_version = f"{status.device.core.version}/{status.device.bridge.version}"
+
+        if fw_version != self._current_fw_version:
+            device_registry = dr.async_get(self.hass)
+            device_entry = device_registry.async_get_device(
+                identifiers={(DOMAIN, status.device.id)}
+            )
+            assert device_entry
+            device_registry.async_update_device(
+                device_entry.id,
+                sw_version=fw_version,
+            )
+            self._current_fw_version = fw_version
```

Comment:
```suggestion
        if self._current_fw_version and fw_version != self._current_fw_version:
            device_registry = dr.async_get(self.hass)
            device_entry = device_registry.async_get_device(
                identifiers={(DOMAIN, status.device.id)}
            )
            assert device_entry
            device_registry.async_update_device(
                device_entry.id,
                sw_version=fw_version,
            )
        self._current_fw_version = fw_version
```
maybe like this? Now it should skip the first update, but update the next ones

Reply:
That works. What do you recommend we do about the sw_paramater in `entity.py`? Set it initially like before with `sw_version=f"{status.device.core.version}/{status.device.bridge.version}",` or make `current_fw_version` public and use this?

Reply:
make it public and use this. Should be set after first update and we only have it in a single place then

Reply:
okay I have pushed a fix with the things we discussed.